app/views/building_map/index.html.erb

```diff
@@ -6,6 +6,6 @@
     },
     polygons: Buildings.transform_leaflet_buildings(Buildings::UNIPOTSDAM_POLYONGS, Buildings::UNIPOTSDAM_STYLING)+
       (Buildings.transform_leaflet_buildings(Buildings::HPI_POLYGONS, Buildings::HPI_STYLING)),
-    polylines: RoutingHelper.transform_route_to_polyline(@route),
+    polylines: @route.present? ? [RoutingHelper.transform_route_to_polyline(@route)] : [],
     markers: Buildings.transform_leaflet_letters(Buildings::HPI_LETTERS) + RoutingHelper.transform_route_to_time_marker(@route))
```

Comment:
This is now inconsistent: For the polyline the check is here, but for time markers it is in the function. Maybe all these lists should be built in a helper so that it can be done in plain ruby?
```ruby
map(
  # ...
  polygons: BuildingMapHelper.leaflet_polygons(),
  polylines: BuildingMapHelper.leaflet_polylines(@route),
  markers: BuildingMapHelper.leaflet_markers(@route),
)
```
... but maybe it's overkill. What do you think?

Reply:
I think it's good and allows scaling.